modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/SourceHandler.java

```diff
@@ -124,10 +127,21 @@ public void requestReceived(NHttpServerConnection conn) {
             if (isMessageSizeValidationEnabled) {
                 httpContext.setAttribute(PassThroughConstants.MESSAGE_SIZE_VALIDATION_SUM, 0);
             }
-
             SourceRequest request = getSourceRequest(conn);
             if (request == null) {
                 return;
+
+            }
+            //check correlationEnabling parameter
+            if (sourceConfiguration.isCorrelationLoggingEnabled()) {
+                setCorrelationId(httpContext, request);
+                long corTime = System.currentTimeMillis();
```

Comment:
Shall we add a meaningful variable name?

Reply:
fixed in https://github.com/wso2/wso2-synapse/pull/1305